common/src/Allocator.h

```diff
@@ -179,7 +179,7 @@ class Allocator {
             }
         }
         
-        assert(chunk != NULL);
+        ensure(chunk != NULL, "chunk is null");
```

Comment:
Please undo this change. The allocator has to be as efficient as possible, and if there's an error in here, we're fucked anyway. Seeing as this code has never caused problems and is used very heavily, it's doubtful that this check would ever fail.